shell/platform/darwin/ios/framework/Source/SemanticsObject.mm

```diff
@@ -0,0 +1,706 @@
+  }
+  [_children release];
+  _children = [[NSMutableArray alloc] initWithArray:children];
+  for (SemanticsObject* child in _children) {
+    [child privateSetParent:self];
+  }
+}
+
+- (void)replaceChildAtIndex:(NSInteger)index withChild:(SemanticsObject*)child {
+  SemanticsObject* oldChild = _children[index];
+  [oldChild privateSetParent:nil];
+  [child privateSetParent:self];
+  [_children replaceObjectAtIndex:index withObject:child];
+}
+
+#pragma mark - UIAccessibility overrides
+
+- (BOOL)isAccessibilityElement {
+  if (![self isAccessibilityBridgeAlive])
+    return false;
+
+  // Note: hit detection will only apply to elements that report
+  // -isAccessibilityElement of YES. The framework will continue scanning the
+  // entire element tree looking for such a hit.
+
+  //  We enforce in the framework that no other useful semantics are merged with these nodes.
+  if ([self node].HasFlag(flutter::SemanticsFlags::kScopesRoute))
+    return false;
+
+  // If the only flag(s) set are scrolling related AND
+  // The only flags set are not kIsHidden OR
+  // The node doesn't have a label, value, or hint OR
+  // The only actions set are scrolling related actions.
+  //
+  // The kIsHidden flag set with any other flag just means this node is now
+  // hidden but still is a valid target for a11y focus in the tree, e.g. a list
+  // item that is currently off screen but the a11y navigation needs to know
+  // about.
+  return (([self node].flags & ~flutter::kScrollableSemanticsFlags) != 0 &&
+          [self node].flags != static_cast<int32_t>(flutter::SemanticsFlags::kIsHidden)) ||
+         ![self node].label.empty() || ![self node].value.empty() || ![self node].hint.empty() ||
+         ([self node].actions & ~flutter::kScrollableSemanticsActions) != 0;
+}
+
+- (void)collectRoutes:(NSMutableArray<SemanticsObject*>*)edges {
+  if ([self node].HasFlag(flutter::SemanticsFlags::kScopesRoute))
+    [edges addObject:self];
+  if ([self hasChildren]) {
+    for (SemanticsObject* child in self.children) {
+      [child collectRoutes:edges];
+    }
+  }
+}
+
+- (BOOL)onCustomAccessibilityAction:(FlutterCustomAccessibilityAction*)action {
+  if (![self node].HasAction(flutter::SemanticsAction::kCustomAction))
+    return NO;
+  int32_t action_id = action.uid;
+  std::vector<uint8_t> args;
+  args.push_back(3);  // type=int32.
+  args.push_back(action_id);
+  args.push_back(action_id >> 8);
+  args.push_back(action_id >> 16);
+  args.push_back(action_id >> 24);
+  [self bridge]->DispatchSemanticsAction([self uid], flutter::SemanticsAction::kCustomAction,
+                                         std::move(args));
+  return YES;
+}
+
+- (NSString*)routeName {
+  // Returns the first non-null and non-empty semantic label of a child
+  // with an NamesRoute flag. Otherwise returns nil.
+  if ([self node].HasFlag(flutter::SemanticsFlags::kNamesRoute)) {
+    NSString* newName = [self accessibilityLabel];
+    if (newName != nil && [newName length] > 0) {
```

Comment:
I'd recommend keeping it.  While true, code that relies on nil handling of messaging is less straight forward.  I wonder if the objc compiler is smart enough to rip it out.

Reply:
This looks like something written by someone who was more familiar with C/C++/Dart/Java rather than ObjC, and is likely to get incorrectly copy-pasted or misunderstood.